tests/e2e/gallery.spec.js

```diff
@@ -8,11 +9,9 @@ test('Test Gallery basic functionalities', async ({page, admin, editor}) => {
     title: 'Test page for Gallery',
     postType: 'page',
   });
-  await editor.canvas.getByRole('button', {name: 'Add default block'}).click();
-  await page.keyboard.type('This is a test Page for gallery.');
-  await page.keyboard.press('Enter');
-  await page.keyboard.type('/gallery');
-  await page.getByRole('option', {name: 'Gallery'}).click();
+
+  // Search block
+  await searchAndInsertBlock({page}, 'Planet 4 Gallery', 'planet4-blocks-gallery');
```

Comment:
Correct me if I'm wrong, but since you provide a namespace here, the block name is not necessary right? You could also write `await searchAndInsertBlock({page}, '', 'planet4-blocks-gallery');`?

Reply:
Nope! leaving a not empty namespace means that the search is more specific. See how it works [here](https://github.com/greenpeace/planet4-master-theme/pull/2419/files#diff-cee58c5a193629b47245c1b3017fb196c8e1c2707a1bd8a384e8948c2c6c7fc5R88)

Reply:
The `blockName` is always required. In terms of typings can be `blockName: string, namespace?: string`

Reply:
If I look at the [function](https://github.com/greenpeace/planet4-master-theme/blob/b2625c0e0731ac92f30286f17cde0b3ee7d4531e/tests/e2e/tools/lib/editor.js#L84-L88), it seems that we don't use the `blockName` when the `namespace` is defined

Reply:
You're correct! I'll replace
`await page.keyboard.type(namespace !== '' ? namespace : blockName);`

to
`await page.keyboard.type(blockName);`

But just take in mind this spec is not going to be merged within this PR. It is been added since the specs here were not passing.

Reply:
In addition, that's why I preferred to define the rest of parameters into an object, and than just pass whatever you need.

const searchAndInsertBlock = async ({page, admin, editor}, {blockName, namespace}) => {